File: crates/flowrt-codegen/tests/golden/sync_fusion_cpp/expected/app/stubs/cpp/sink.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Sink final : public flowrt_app::SinkInterface {
public:
    flowrt::Status on_tick(
        const flowrt::Latest<flowrt_app::Estimate>& estimate) override {
        (void)estimate;
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Sink>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/sync_fusion_rust/expected/app/stubs/rust/sink.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Sink;

impl flowrt_app::components::Sink for Sink {
    fn on_tick(
        &mut self,
        estimate: flowrt::Latest<'_, flowrt_app::messages::Estimate>,
    ) -> flowrt::Status {
        let _ = estimate;
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Sink::default()))
}
